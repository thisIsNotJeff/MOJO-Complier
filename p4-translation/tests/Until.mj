{
    var x := 0;
    var y := 1;

    loop while x < 16 {
        x := x + 1;
        y := y + y;
        if y == 42 {
            break;
        }
    } until y == 512;
}
