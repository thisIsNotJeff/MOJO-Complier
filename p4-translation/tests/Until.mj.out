	.text
Until:
MOVE(
 TEMP Until.1.x,
 CONST 0),
MOVE(
 TEMP Until.1.y,
 CONST 1),
JUMP(
 NAME L.1),
LABEL L.2,
MOVE(
 TEMP Until.1.x,
 ADD(
  TEMP Until.1.x,
  CONST 1)),
MOVE(
 TEMP Until.1.y,
 ADD(
  TEMP Until.1.y,
  TEMP Until.1.y)),
BEQ(
 TEMP Until.1.y,
 CONST 42,
 L.4, L.5),
LABEL L.4,
JUMP(
 NAME L.3),
LABEL L.5,
BEQ(
 TEMP Until.1.y,
 CONST 512,
 L.3, L.1),
LABEL L.1,
BLT(
 TEMP Until.1.x,
 CONST 16,
 L.2, L.3),
LABEL L.3


